spec/controllers/conference_registration_controller_spec.rb

```diff
@@ -1,51 +1,179 @@
+    end
 
     describe 'GET #new' do
       context 'registration period open' do
-        before do
-          @registration_period = create(:registration_period, conference: conference)
+        before :each do
+          create(:registration_period, conference: conference, start_date: 3.days.ago, end_date: 1.day.from_now)
         end
 
-        context 'user registered' do
-          before do
-            @registration = create(:registration, conference: conference, user: user)
-            get :new, conference_id: conference.short_title
+        context 'registration limit not exceeded' do
+          before :each do
+            conference.registration_limit = 0
+            conference.save!
           end
 
-          it 'redirects to edit conference registration' do
-            expect(response).to redirect_to edit_conference_conference_registration_path(conference.short_title)
+          context 'OSEM_ICHAIN_ENABLED is true' do
+            before :each do
+              stub_const('ENV', ENV.to_hash.merge('OSEM_ICHAIN_ENABLED' => 'false'))
+            end
+
+            context 'user registered' do
+              it_behaves_like 'access action without ability', :registered_user, 'true', '/conferences/myconf/register/edit', nil
+            end
+
+            context 'user not registered' do
+              before :each do
+                get :new, conference_id: conference.short_title
+              end
+
+              it 'user variable exists' do
+                expect(assigns(:user)).not_to be_nil
+              end
+
+              it 'renders the new template' do
+                expect(response).to render_template('new')
+              end
+            end
+          end
+
+          context 'OSEM_ICHAIN_ENABLED is false' do
+            before :each do
+              stub_const('ENV', ENV.to_hash.merge('OSEM_ICHAIN_ENABLED' => 'false'))
+            end
+
+            context 'user registered' do
+              it_behaves_like 'access action without ability', :registered_user, 'false', '/conferences/myconf/register/edit', nil
+            end
+
+            context 'user not registered' do
+              before :each do
+                get :new, conference_id: conference.short_title
+              end
+
+              it 'user variable exists' do
+                expect(assigns(:user)).not_to be_nil
+              end
+
+              it 'renders the new template' do
+                expect(response).to render_template('new')
+              end
+            end
           end
         end
 
-        context 'user not registered' do
-          before do
-            get :new, conference_id: conference.short_title
+        context 'registration limit exceeded' do
+          before :each do
+            conference.registration_limit = 1
+            conference.save!
           end
 
-          it 'user variable exists' do
-            expect(assigns(:user)).not_to be_nil
+          context 'OSEM_ICHAIN_ENABLED is true' do
```

Comment:
This context does not really offer any value here, specially since there is no before block anymore. You can just include the info that ichain is true in the 2 contexts below (for register and not registered user).